constraint/pkg/client/client.go

```diff
@@ -619,10 +623,55 @@ func (c *Client) RemoveData(ctx context.Context, data interface{}) (*types.Respo
 	return resp, &errMap
 }
 
-// Review makes sure the provided object satisfies all stored constraints.
+// Review makes sure the provided object satisfies constraints applicable for specific enforcement points.
 // On error, the responses return value will still be populated so that
 // partial results can be analyzed.
-func (c *Client) Review(ctx context.Context, obj interface{}, opts ...drivers.QueryOpt) (*types.Responses, error) {
+func (c *Client) Review(ctx context.Context, obj interface{}, opts ...reviews.ReviewOpt) (*types.Responses, error) {
+	// Initialize an empty slice for enforcement points
+	var eps []string
+	// Create a new ReviewCfg instance
+	cfg := &reviews.ReviewCfg{}
+	// Apply each option to the ReviewCfg instance
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	// Check if a source enforcement point is specified in the configuration
+	if cfg.SourceEP != "" {
+		// Initialize enforceAll as false. It will be used to determine if all enforcement points should be enforced
+		enforceAll := false
+		// Iterate through the client's enforcement points
+		for _, ep := range c.enforcementPoints {
+			// Check if the current enforcement point indicates all enforcement points should be enforced
+			if ep == apiconstraints.AllEnforcementPoints {
+				enforceAll = true
+			}
+			// If the specified source enforcement point matches the current enforcement point, add it to the list
+			if cfg.SourceEP == ep {
+				eps = append(eps, ep)
+				break // Exit the loop since the matching enforcement point is found
+			}
+		}
+		// If enforceAll is true, add the source enforcement point to the list of enforcement points
+		if enforceAll {
+			eps = append(eps, cfg.SourceEP)
```

Comment:
what if cfg.SourceEP is already appended?

Reply:
nvm I see it in GK

Reply:
Fixed it